src/bluesearch/database/article.py

```diff
@@ -227,24 +215,78 @@ def get_paragraphs_sections_mapping(self) -> Dict[Element, str]:
 
         return mapping
 
-    @staticmethod
-    def text_content(element: Optional[Element]) -> str:
-        """Extract all text of an element and of its descendants (at any depth).
+    def _inner_text(self, element: Element) -> str:
+        """Convert all inner text and sub-elements to one string.
+
+        In short, we collect all the inner text while also converting all
+        sub-elements that we encounter to strings using ``self._element_to_str``.
+        All escaped HTML in the raw text is unescaped.
+
+        For example, if schematically the element is given by
+
+            element = "<p>I <bold>like</bold> python &amp; ice cream.<p>"
+
+        then ``_inner_text(element)`` would give
+
+            "I like python & ice cream."
+
+        provided that "<bold>like</bold>" is resolved to "like" by the
+        ``self._element_to_str`` method.
```

Comment:
Yes, good question. That's because internally it calls `self._element_to_str`. Which enables the whole custom tag processing. I added a test in ff67c8d that would not pass.

Reply:
I see.